File: apps/api/momento_api/models/tag.py
from pydantic import BaseModel


class TagResponse(BaseModel):
    id: int
    name: str
    created_at: str


class TagCreateRequest(BaseModel):
    name: str


class TagDeleteRequest(BaseModel):
    tag_id: int


class TagAddToMediaRequest(BaseModel):
    tag_id: int
    media_ids: list[int]


class TagRemoveFromMediaRequest(BaseModel):
    tag_id: int
    media_ids: list[int]


class TagListResponse(BaseModel):
    tags: list[TagResponse]

File: src/backend/models/tag.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TagResponse {
    pub id: i64,
    pub name: String,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TagCreateRequest {
    pub name: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TagDeleteRequest {
    pub tag_id: i64,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TagAddToMediaRequest {
    pub tag_id: i64,
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TagRemoveFromMediaRequest {
    pub tag_id: i64,
    pub media_ids: Vec<i64>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct TagListResponse {
    pub tags: Vec<TagResponse>,
}
